app/javascript/onboarding/components/ProfileForm.jsx

```diff
@@ -1,55 +1,50 @@
-import { h, Component } from 'preact';
+import { h } from 'preact';
 import PropTypes from 'prop-types';
-
+import { useEffect, useState } from 'preact/hooks';
 import { userData, getContentOfToken, updateOnboarding } from '../utilities';
 import Navigation from './Navigation';
+import OnboardingForm from './OnboardingForm';
+import OnboardingContentHeader from './OnboardingContentHeader';
 
-/* eslint-disable camelcase */
-class ProfileForm extends Component {
-  constructor(props) {
-    super(props);
-
-    this.handleChange = this.handleChange.bind(this);
-    this.onSubmit = this.onSubmit.bind(this);
-    this.user = userData();
+const lastOnboardingPage = 'v2: personal info form';
 
-    this.state = {
-      formValues: {
-        summary: '',
-        location: '',
-        employment_title: '',
-        employer_name: '',
-      },
-      last_onboarding_page: 'v2: personal info form',
-      canSkip: true,
-    };
-  }
+const ProfileForm = ({
+  prev,
+  slidesCount,
+  currentSlideIndex,
+  communityConfig,
+  next,
+}) => {
+  const [formValues, setFormValues] = useState({
+    summary: '',
+    location: '',
+    employment_title: '',
+    employer_name: '',
+  });
+  const [canSkip, setCanSkip] = useState(true);
 
-  componentDidMount() {
-    updateOnboarding('v2: personal info form');
-  }
+  useEffect(() => {
+    updateOnboarding(lastOnboardingPage);
+  }, []);
 
-  onSubmit() {
+  const onSubmit = () => {
```

Comment:
suggestion (non-blocking): Consider using the request utility function, https://github.com/forem/forem/blob/master/app/javascript/utilities/http/request.js, instead of fetch. It handles the content type and CSRF token plus other sensible defaults for the application out of the box.